ddtrace/contrib/dbapi/__init__.py

```diff
@@ -28,52 +28,39 @@ def __init__(self, cursor, pin):
         name = pin.app or 'sql'
         self._datadog_name = '%s.query' % name
 
-    def executemany(self, query, *args, **kwargs):
+    def _trace_method(self, method, resource, extra_tags, *args, **kwargs):
         pin = self._datadog_pin
         if not pin or not pin.enabled():
-            return self.__wrapped__.executemany(query, *args, **kwargs)
+            return method(*args, **kwargs)
         service = pin.service
 
-        # FIXME[matt] properly handle kwargs here. arg names can be different
-        # with different libs.
-        with pin.tracer.trace(self._datadog_name, service=service, resource=query) as s:
+        with pin.tracer.trace(self._datadog_name, service=service, resource=resource) as s:
             s.span_type = sql.TYPE
-            s.set_tag(sql.QUERY, query)
+            s.set_tag(sql.QUERY, resource)
             s.set_tags(pin.tags)
-            s.set_tag("sql.executemany", "true")
-            try:
-                return self.__wrapped__.executemany(query, *args, **kwargs)
-            finally:
-                s.set_metric("db.rowcount", self.rowcount)
 
-    def execute(self, query, *args, **kwargs):
-        pin = self._datadog_pin
-        if not pin or not pin.enabled():
-            return self.__wrapped__.execute(query, *args, **kwargs)
+            for k, v in extra_tags.items():
```

Comment:
yes! clearing the code as we refactor is great! unfortunately we've many integrations that are not up to date with newer API! Thanks for spotting and fixing all these inconsistencies!